packages/editor/src/lib/utils/reorderShapes.ts

```diff
@@ -155,19 +153,18 @@ function reorderToFront(moving: Set<TLShape>, children: TLShape[], changes: TLSh
 }
 
 function getOverlapChecker(editor: Editor, moving: Set<TLShape>) {
```

Comment:
We don't really need to check all of the vertices, since this is a minor optimization anyway. Bounds should be fine.